### PR TITLE
Volunteer signup

### PR DESCRIPTION
**Pull request by sohak22**
Notion ticket link:

https://www.notion.so/uwblueprintexecs/FRONTEND-Volunteer-Participant-Info-Sign-Up-Page-2efff449c0aa4819a3abbd4a5b936d6d

### Implementation description

- Front end volunteer sign up form, with capabilities to save fields in local storage

### Steps to test

1. localhost:3000/volunteersignup

### Checklist

-   [ ] My PR name is descriptive and in imperative tense
-   [ ] I have run the linter
-   [ ] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/volunteersignup.tsx [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+} from "@chakra-ui/react";
+import { CloseButton } from "@components/CloseButton";
+import { BackButton } from "@components/BackButton";
+import { useState } from "react";
+import { GetServerSideProps } from "next"; // Get server side props
+import { getSession, GetSessionOptions } from "next-auth/client";
+import Wrapper from "@components/SDCWrapper";
+import useLocalStorage from "@utils/useLocalStorage";
+
+const BLUE = "#0C53A0"; // TODO: move to src/styles
```
Comment (mahadzaryab1) [inline pages/volunteersignup.tsx]: flagging this TODO

File: pages/volunteersignup.tsx [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+}
+
+/**
+ * getServerSideProps gets the session before this page is rendered
+ */
+export const getServerSideProps: GetServerSideProps = async (
+    context: GetSessionOptions,
+) => {
+    // obtain the next auth session
+    const session = await getSession(context);
```
Comment (mahadzaryab1) [inline pages/volunteersignup.tsx]: what exactly is the session being used for in this page?

File: pages/volunteersignup.tsx [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+                            _focus={{}}
+                            href="/"
+                        >
+                            <Button
+                                color={"white"}
+                                bg={"#0C53A0"}
+                                px={10}
+                                _hover={{
+                                    bg: "#2C6AAD",
+                                }}
```
Comment (mahadzaryab1) [inline pages/volunteersignup.tsx]: colors can be moved to `src/styles`

**Review by mahadzaryab1 — COMMENTED** [review]
Added a couple of comments/questions. Also let's rename the page to something else - `volunteers/signup` maybe?

File: pages/volunteersignup.tsx (hunk 2 above)
Comment (sohak22) [reply pages/volunteersignup.tsx]: The session is for the wrapper

File: pages/volunteersignup.tsx [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+    const [city, setCity] = useLocalStorage("city", "");
+    /*   const [participantProvince, setParticipantProvince] =
+          useState(DEFAULT_PROVINCE); */
+    const [postalCode, setPostalCode] = useLocalStorage("postalCode", "");
+    const [school, setSchool] = useLocalStorage("school", "");
+
+    //volunteer personal details
+
+    const [skills, setSkills] = useLocalStorage("skills", "");
+    const [hear, setHear] = useLocalStorage("hear", "");
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: can you maybe change this to `heardFrom` or something more descriptive

File: pages/volunteersignup.tsx [hunk 5]
```diff
@@ -0,0 +1,481 @@
...
+                </FormControl>
+            </FormPage>
+            <FormButton onClick={formButtonOnClick}>Next</FormButton>
+        </Box>,
+        <Box>
+            <FormPage>
+                <VStack>
+                    <FormControl id="skills">
+                        <FormLabel>
+                            {" "}
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: what's this / is it needed

File: pages/volunteersignup.tsx [hunk 6]
```diff
@@ -0,0 +1,481 @@
...
+                            Skills/Experience (ex. Arts and Crafts, Music,
+                            First-Aid Certificates, Teaching or Volunteering
+                            Experience, Experience with Children with Special
+                            Needs)
+                        </FormLabel>
+                        <Textarea placeholder="Type here" size="sm"></Textarea>
+                    </FormControl>
+                    <FormControl id="hear-about-us">
+                        <FormLabel>
+                            {" "}
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: same as above

File: pages/volunteersignup.tsx [hunk 7]
```diff
@@ -0,0 +1,498 @@
...
+    const [volunteerLastName, setVolunteerLastName] = useLocalStorage(
+        "volunteerLastName",
+        "",
+    );
+    const [phoneNumber, setPhoneNumber] = useLocalStorage("phoneNumber", "");
+
+    //volunteer Personal Details
+
+    const [dateOfBirth, setDateOfBirth] = useLocalStorage("dateOfBirth", "");
+    const [fifteen, setFifteen] = useState(false);
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: doesn't look like this is used anywhere yet

**Review by jason17h — COMMENTED** [review]
Looking good so far, just a few comments + I added some changes so be sure to pull!!

File: pages/volunteersignup.tsx [hunk 8]
```diff
@@ -0,0 +1,494 @@
...
+import { useState } from "react";
+import { GetServerSideProps } from "next"; // Get server side props
+import { getSession, GetSessionOptions } from "next-auth/client";
+import Wrapper from "@components/SDCWrapper";
+import useLocalStorage from "@utils/useLocalStorage";
+import { roles, locale, province, VolunteerInput } from "@models/User";
+
+const BLUE = "#0C53A0"; // TODO: move to src/styles
+const RADIO_YES = "yes";
+const RADIO_NO = "no";
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: looks like `RADIO_YES` and `RADIO_NO` are not used anymore so they can be removed. same with some of the imports I think

File: pages/volunteersignup.tsx [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+    const [participantProvince, setParticipantProvince] =
+        useState(DEFAULT_PROVINCE);
+    const [postalCode, setPostalCode] = useLocalStorage("postalCode", "");
+    const [school, setSchool] = useLocalStorage("school", "");
+
+    //volunteer personal details
+
+    const [skills, setSkills] = useLocalStorage("skills", "");
+    const [heardFrom, setHeardFrom] = useLocalStorage("heardFrom", "");
+    const [attending, setAttending] = useState(false);
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: these don't look like they're used - should they be used somewhere? or can we get rid of them? it looks like they aren't needed

**Review by jason17h — APPROVED** [review]
just some minor comments to remove unused things but other than that looks good to me (should get at least 1 more approval though)! Thanks for doing this 😎

File: pages/volunteersignup.tsx (hunk 9 above)
Comment (sohak22) [reply pages/volunteersignup.tsx]: Wait im not sure which one your talking about but Skills is used on line 238-239, heard from is used under that on lines 249-250. And I took out attending

File: src/style/colours.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import { extendTheme } from "@chakra-ui/react";
+
+const colours = extendTheme({
+    colors: {
+        Blue: "#0C53A0",
+    },
+});
+
+export default colours;
```
Comment (jason17h) [inline src/style/colours.ts]: can you move this to `src/styles` instead of `src/style`

File: pages/volunteersignup.tsx [hunk 11]
```diff
@@ -0,0 +1,492 @@
...
+                            _hover={{ textDecoration: "none" }}
+                            _focus={{}}
+                            href="/"
+                        >
+                            <Button
+                                color={"white"}
+                                bg={colourTheme.colors.Blue}
+                                px={10}
+                                _hover={{
+                                    bg: colourTheme.colors.LightBlue,
```
Comment (jason17h) [inline pages/volunteersignup.tsx]: build is failing because there is no `LightBlue` defined

File: pages/volunteersignup.tsx (hunk 9 above)
Comment (jason17h) [reply pages/volunteersignup.tsx]: I was just referring to attending

**Review by jason17h — APPROVED** [review]

**Review by feng-eric — APPROVED** [review]

**Comment by jason17h** [comment]
Btw @sohak22 can you update the signup page to redirect to this form if you click the "Volunteer" button? https://github.com/uwblueprint/social-diversity-for-children/blob/71edf4e5961fd4d76680b4f998e45bb78eae3da8/pages/signup.tsx#L54

**Review by feng-eric — COMMENTED** [review]
Sorry, thought I submitted the review a few days ago. Had one small change I wanted to comment on.

**Review by feng-eric — APPROVED** [review]
LGTM, ping me before you merge because we need to reset the database since we're making DB schema changes.

**Pull request merged**